lib/smart_proxy_openscap/arf_parser.rb

```diff
@@ -13,17 +13,13 @@ def initialize(cname, policy_id, date)
       end
 
       def as_json(arf_data)
-        begin
-          file = Tempfile.new
+        decompressed = Tempfile.create do |file|
           file.write(arf_data)
-          file.rewind
-          decompressed = `bunzip2 -dc #{file.path}`
+          file.flush
```

Comment:
Is there a reason to use flush instead of rewind? I mean, I don't think we even needed rewind, since the file should be opened by a different process, thus the different stream?..

Reply:
Personal preference I guess, for me `flush` better declares what we want to do. Flushing writes the bits to disk, rewinding does the same thing plus some additional things that we don't really care about.

Reply:
It seems it's just an OCD "fix", Ruby doesn't necessary does that "really": https://ruby-doc.org/core-2.5.5/IO.html#method-i-flush. Just in Ruby world.

Reply:
https://ruby-doc.org/core-2.5.5/IO.html#method-i-rewind doesn't even say that data is written to disk while that is exactly what we want.

Looking at https://www.man7.org/linux/man-pages/man3/rewind.3p.html also doesn't make any mention of writing to disk while I think flush maps to [fflush](https://www.man7.org/linux/man-pages/man3/fflush.3.html) and is exactly what is needed in this case.

Reply:
Right, I might have oversimplified it. But the general idea still stands, no? We need to call flush or rewind so that the next process can read it. Out of the two, one expresses the intent more cleanly?

Reply:
Flush. I was trying to say that I'm not sure if rewind actually guarantees it's flushed and wondering how it worked in the first place

Reply:
Just another thought: instead of writing it to a temporary file you can perhaps also present it as stdin to the process by passing `:stdin_data` as an option to `Open3.capture3`. Not insisting, but it's another option.